src/components/Container.js

```diff
@@ -64,9 +67,9 @@ class Container extends React.Component {
 
   drawEnd(e) {
     const { feature } = e;
-    const { interaction, selectedTrail, trails, addTrail, addHydrant, modifyTrail } = this.props;
+    const { interaction, selectedTrail, editableTrail, trails, addTrail, addHydrant, modifyTrail } = this.props;
     if (interaction === 'DRAW_MODIFY_TRAIL') {
-      const trail = trails.get(selectedTrail);
+      const trail = trails.get(editableTrail);
```

Comment:
I think having editableTrail potentially be different from selectedTrail could lead to weird states we don't want, either in components themselves or in the map-features updating, plus it's redundant since it will always either be same as selectedTrail or null. Clearer to have a boolean for "isEditing" or "isEditable" or something like that which applies to the selectedTrail.

Reply:
Check